controllers/managedocs_controller_test.go

```diff
@@ -434,11 +434,96 @@ var _ = Describe("ManagedOCS controller", func() {
+				Skip("Skipping this test till Nooba MCG is integrated in deployer")
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secret.Data["size"] = []byte("1")
+				Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
+				// Ensure, over a period of time, that the resource is not created
+				utils.EnsureNoResource(k8sClient, ctx, scTemplate.DeepCopy(), timeout, interval)
+			})
+		})
+		When("there is an invalid enable-mcg value in the add-on parameters secret", func() {
+			It("should not create reconciled resources", func() {
+				// Create a invalid enable-mcg parameter value
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secret.Data["size"] = []byte("1")
+				secret.Data["enable-mcg"] = []byte("NO")
+				Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
+
+				// Ensure, over a period of time, that the resource is not created
+				utils.EnsureNoResource(k8sClient, ctx, scTemplate.DeepCopy(), timeout, interval)
+			})
+		})
+		When("there is a valid enable-mcg value in the addon-on parameters secret which is false", func() {
+			It("should not change the storage cluster spec", func() {
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secret.Data["size"] = []byte("1")
+				secret.Data["enable-mcg"] = []byte("false")
+				Expect(k8sClient.Update(ctx, secret)).Should(Succeed())
+
+				Eventually(func() bool {
+					sc := scTemplate.DeepCopy()
+					err := k8sClient.Get(ctx, utils.GetResourceKey(sc), sc)
+					if err != nil {
+						return true
+					}
+					return sc.Spec.MultiCloudGateway.ReconcileStrategy == "manage"
+				}, timeout, interval).Should(BeFalse())
+			})
+		})
+		When("there is a valid enable-mcg value in the addon-on parameters secret which is true", func() {
+			It("should change the storage cluster spec to enable MCG", func() {
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secret.Data["size"] = []byte("1")
+				secret.Data["enable-mcg"] = []byte("true")
+				Expect(k8sClient.Update(ctx, secret)).Should(Succeed())
+
+				Eventually(func() bool {
+					sc := scTemplate.DeepCopy()
+					err := k8sClient.Get(ctx, utils.GetResourceKey(sc), sc)
+					if err != nil {
+						return false
+					}
+					return sc.Spec.MultiCloudGateway.ReconcileStrategy == "manage"
+				}, timeout, interval).Should(BeTrue())
+			})
+		})
+		When("MCG is already enabled and enable-mcg value in addon-on parameter secret is false", func() {
+			It("should not disable MCG in storage cluster spec", func() {
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secret.Data["enable-mcg"] = []byte("false")
+				Expect(k8sClient.Update(ctx, secret)).Should(Succeed())
+				// MCG spec should not get updated
+				Eventually(func() bool {
+					sc := scTemplate.DeepCopy()
+					err := k8sClient.Get(ctx, utils.GetResourceKey(sc), sc)
+					if err != nil {
+						return false
+					}
+					return sc.Spec.MultiCloudGateway.ReconcileStrategy == "manage" ||
+						sc.Spec.MultiCloudGateway.ReconcileStrategy == ""
```

Comment:
`""` is not an acceptable value in this case, we need to look specifically for "manage"